src/git.rs

```diff
@@ -129,23 +129,30 @@ impl GitHelper {
         last_version: &SemVer,
         prerelease: &semver::Prerelease,
     ) -> Option<semver::Prerelease> {
-        self.version_map
+        let mut prereleases = self
+            .version_map
             .values()
             .flat_map(|vat| vat.iter())
             .map(|vat| &vat.version.0)
             .filter(|version| {
                 version.major == last_version.major()
                     && version.minor == last_version.minor()
                     && version.patch == last_version.patch()
+                    && version
+                        .pre
+                        .rsplit_once('.')
+                        .filter(|pre| prerelease.as_str() == pre.0)
+                        .is_some()
             })
-            .find(|version| {
-                version
-                    .pre
-                    .rsplit_once('.')
-                    .filter(|pre| prerelease.as_str() == pre.0)
-                    .is_some()
-            })
-            .map(|v| v.pre.clone())
+            .collect::<Vec<_>>();
+
+        println!("{:?}", prereleases);
```

Comment:
This println can be removed. 
Another fix could be to make VersionAndTag a BTreeMap instead of a hashmap. 
Thanks for the PR.

Reply:
I updated the commit a little bit and will merge it soon. 
A btreemap would not solve anything as the key is the oid.

I did however sort normally and take the last version, which resembles more finding the last version.